bears/python/PycodestyleBear.py

```diff
@@ -61,3 +72,26 @@ def create_arguments(
         arguments.append(filename)
 
         return arguments
+
+    def process_output(
+            self,
+            output,
+            filename,
+            file,
+            ):
+        result = re.match(OUTPUT_REGEX, output)
+        if not result:
+            return
+        line, column, message, origin = result.groups()
+        if origin == 'E501':
+            aspect = LineLength('py')
+        else:
+            aspect = None
+        yield Result.from_values(
+            origin=self,
```

Comment:
this is a regression.
Previously the origin was the error code.
you are discarding that

Reply:
origin is to be kept `PycodestyleBear` only as that when we want to ignore PycodestyleBear, we have to do `# Ignore Pycodestyle` and not `# Ignore E501(or any other rule)`.

I have changed it to `rule` so that there's no confusion.

Reply:
it was previously `origin` ... was it being ignored previously?

if the `Result` objects were previously setting origin to `PycodestyleBear`, then why was the regex capturing `origin` .. what was happening with it previously?

wrt `# Ignore ...` if the bear was previously using `origin` of `E501`, then users *must* be using `# Ignore E501` currently, and if you change it then their CI will break.

So we need to be careful.

We can still change it, but we need to ensure that change isnt hidden in an 'aspectification' commit.  It needs to be a separate issue which is classified as a bug, and properly analysed.

Reply:
Currently, it uses `# Ignore Pycodestyle` or `# Ignore Pycodestyle (E501)` (an example is https://github.com/coala/coala-bears/blob/e46d352d30f5895b4deead2bcc174a50f480aade/tests/js/PrettierLintBearTest.py#L11).

I see what `@linter` was doing previously, see https://github.com/coala/coala/blob/d936d1624a8d37b4be96d023079c39f244a351c3/coalib/bearlib/abstractions/Linter.py#L320

Now I have changed this to what was being done previously in the linter.

Reply:
Thank you -- that is better.